vault/external_tests/kv/kv_subkeys_test.go

```diff
@@ -93,19 +100,31 @@ func TestKV_Subkeys_Deleted(t *testing.T) {
 		},
 	}
 
-	resp, err := c.Logical().Write("kv/data/foo", kvData)
+	resp, err := kvRequestWithRetry(t, func() (interface{}, error) {
+		return c.Logical().Write("kv/data/foo", kvData)
+	})
 	if err != nil {
 		t.Fatalf("write failed, err :%v, resp: %#v", err, resp)
 	}
 
-	resp, err = c.Logical().Delete("kv/data/foo")
+	secretRaw, err := kvRequestWithRetry(t, func() (interface{}, error) {
+		return c.Logical().Delete("kv/data/foo")
+	})
 	if err != nil {
-		t.Fatalf("delete failed, err :%v, resp: %#v", err, resp)
+		t.Fatalf("delete failed, err :%v, resp: %#v", err, secretRaw)
 	}
 
-	req := c.NewRequest("GET", "/v1/kv/subkeys/foo")
-	apiResp, err := c.RawRequestWithContext(context.Background(), req)
-	if resp != nil {
+	apiRespRaw, err := kvRequestWithRetry(t, func() (interface{}, error) {
+		req := c.NewRequest("GET", "/v1/kv/subkeys/foo")
```

Comment:
this is a q more for consistency's sake, but would it be possible to do c.Logical().Read(..) here like in the other get calls ? I saw the PATCH calls are also like this and assumed there isn't a `c.Logical().Patch()` function, but not sure if the same logic applies here.

Reply:
Using `RawRequestWithContext` directly is necessary in these cases so that we have access to the response status code. Using the helper functions like `Read`, `Write`, etc. do not provide that information.

Reply:
aah gotcha!!